Source/Read/SourceControl/GitHub/RepositoriesEventProcessor.cs

```diff
@@ -24,60 +24,58 @@ IReadModelRepositoryFor<InstallationRepositories> repositoryForInstallationRepos
             _repositoryForRepositoriesList = repositoryForRepositoriesList;
             _repositoryForInstallationsList = repositoryForInstallationsList;
             _repositoryForInstallationRepositories = repositoryForInstallationRepositories;
-        }
 
-        // TODO: How do we do this?
-        readonly static SemaphoreSlim _lock = new SemaphoreSlim(1,1);
```

Comment:
I see you're removing this; but since you're asking "How do we do this?"..

You could just do:

```csharp
readonly static object _lock = new object();
```

In a method: 

```csharp
lock( _lock ) 
{
    // do stuff
}
```

Reply:
Well, it's not removed, not sure why it looks like that...
Anyways the question is not so much about the locking. I think I copied it from somewhere I used `await`, but I should definitely have a look and make sure it's released.

__The question is more about "How do we deal with concurrent commands?". I guess the answer is that in most cases we just let one of the commands pass, and the rest would fail?__

In this case, it shouldn't really fail because GitHub won't retry the webhook. But I guess I could do the queuing a bit closer to the webhook-specific code so that it doesn't pollute the normal DDD code.

Reply:
I'm pretty sure I answered my own question.